include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -1856,7 +1880,23 @@ struct MinMaxConverter : public OpRewritePattern<CmpOp> {
     auto selectOp =
         dyn_cast<arith::SelectOp>(*cmpOp.getResult().getUsers().begin());
     if (!selectOp) {
-      return failure();
+      // Try to find a select op consuming an `ori` op that consumes the cmpOp.
+      selectOp = findSelectOpConsumingCmp(cmpOp.getResult());
+      if (!selectOp) {
+        return failure();
+      }
+      // Now, we have found a select op consuming an `ori` op that consumes the
+      // cmpOp. We can proceed to check if the select op is part of the same
+      // reduce op.
+      auto reduceOp = findEnclosingReduceOp(selectOp);
+      if (reduceOp != findEnclosingReduceOp(cmpOp)) {
```

Comment:
this pattern doesn't require us to be inside a reduce op, so i think we can relax this condition